cmd/kubectl-gadget/deploy.go

```diff
@@ -506,12 +546,26 @@ func runDeploy(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
+	// modify the otel-metrics-listen-address if specified
+	if otelMetricsListenAddr != "" {
+		// iterate over the objects and find the DaemonSet
+		// modify the prometheus.io/port annotation & break
+		for objectNum, obj := range objects {
+			if isDaemonSet(obj) {
+				demonObject, err := modifyPrometheusPort(objects[objectNum], otelMetricsListenAddr)
```

Comment:
I think we should only specify the port here so we can use something [SplitHostPort ](https://pkg.go.dev/net#SplitHostPort) to get the port only here?